recipe/meta.yaml

```
@@ -1,41 +1,142 @@
+      - lib/libopenjp*.dylib                                # [osx]
+      - lib/libopenjp*.so                                   # [linux]
+      - lib/libopenjp*.so.*                                 # [linux]
+      - lib/pkgconfig/libopenjp*.pc                         # [linux or osx]
+      - lib/cmake/{{ name }}-{{ short_version }}            # [linux or osx]
+      - include/{{ name }}-{{ short_version }}              # [linux or osx]
+      - Library/lib/pkgconfig/libopenjp*.pc                 # [win]
+      - Library/lib/cmake/{{ name }}-{{ short_version }}    # [win]
+      - Library/include/{{ name }}-{{ short_version }}      # [win]
+      - Library/bin/openjp*.dll                             # [win]
+      - Library/lib/openjp*.lib                             # [win]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libopenjp2.{{ version }}.dylib    # [osx]
+        - test -f $PREFIX/lib/libopenjp2.so.{{ version }}       # [linux]
+        - if not exist "%LIBRARY_BIN%\openjp2.dll" exit 1       # [win]
+    about:
+        home: https://www.openjpeg.org/
+        dev_url: https://github.com/uclouvain/openjpeg
+        doc_url: https://github.com/uclouvain/openjpeg/wiki/DocJ2KCodec
+        license: BSD-2-Clause
+        license_family: BSD
+        license_file: LICENSE
+        summary: An open-source JPEG 2000 codec written in C.
+        description: |
+          OpenJPEG is an open-source JPEG 2000 codec written in C language.
+          It has been developed in order to promote the use of JPEG 2000,
+          a still-image compression standard from the Joint Photographic Experts Group (JPEG).
+          Since may 2015, it is officially recognized by ISO/IEC and ITU-T
+          as a JPEG 2000 Reference Software.
+
+          This package provides the OpenJPEG library and its development headers.
+
+  - name: {{ name }}
+    run_exports:
+      # good compatibility in 2.x series, check before new release
+      # http://www.openjpeg.org/abi-check/timeline/openjpeg/
+      - {{ pin_subpackage('lib' + name, max_pin='x.x') }}
+    ignore_run_exports:
+      # Needed as a transitive build-time dependency but is not actually required by the output binaries.
+      - zlib
+    requirements:
+      build:
+        - {{ stdlib('c') }}
+        - {{ compiler('c') }}
+      host:
+        - {{ pin_subpackage('lib' + name, exact=True) }}
+        # Refer to the package changelog for the latest versions of the dependencies
+        # https://github.com/uclouvain/openjpeg/blob/master/CHANGELOG.md
+        # libtiff: package changelog indicates 4.6.0, but we don't have this version in our catalog at this time
+        - libtiff 4.7.0
+        # libpng: package changelog indicates 1.6.43, but we don't have this version in our catalog at this time
+        - libpng 1.6.39
+        - zlib 1.3.1
+        # liblcms2: package changelog indicates 2.8, but we don't have this version in our catalog at this time
+        - lcms2 2.16
```

```suggestion
        - libtiff {{ libtiff }}
        # libpng: package changelog indicates 1.6.43, but we don't have this version in our catalog at this time
        - libpng {{ libpng }}
        - zlib 1.3.1
        # liblcms2: package changelog indicates 2.8, but we don't have this version in our catalog at this time
        - lcms2 {{ lcms2 }}
```

For maintainability reasons, replace `foo` with `foo {{ foo }}` in the `host` section if the  package name is present in https://github.com/AnacondaRecipes/aggregate/blob/master/conda_build_config.yaml. 

Our current default zlib is 1.2 https://github.com/AnacondaRecipes/aggregate/blob/master/conda_build_config.yaml#L884 but libzlib is 1 https://github.com/AnacondaRecipes/aggregate/blob/master/conda_build_config.yaml#L636. Will it be compatible with other packages at runtime?

---

Re: `zlib`, the package has v1.3.1 in their `thirdparty` directory: https://github.com/uclouvain/openjpeg/commit/060caa9308ea8293e7743c998db2d7475c2e84d4

That makes me think a little bit more broadly: Would a JPEG file generated by the Anaconda `openjpeg` package depending on `zlib 1.2` be compatible with the JPEG files generated by `openjpeg` with `zlib v1.3.1` dependency? There is [a StackOverflow answer](https://stackoverflow.com/a/70642499) by one of the `zlib` maintainers mentioning that there would be no compatibility issues. Also, [zlib documentation](https://zlib.net/manual.html#:~:text=ZEXTERN%20const%20char%20*%20ZEXPORT%20zlibVersion,made%20by%20deflateInit%20and%20inflateInit.) says it would check the "first character" and I believe that corresponds to the major version number "1".

I am guessing replacing `1.3.1` to `{{ zlib }}` and using `zlib 1.2` would be okay from `openjpeg` package point of view.